Dockerfile

```diff
@@ -1,31 +1,64 @@
 # syntax=docker/dockerfile:1
 
-# FINAL docker image for music assistant server
-# This image is based on the base image and installs
-# the music assistant server from our built wheel on top.
-
+# Builder image. It builds the venv that will be copied to the final image
+# 
 ARG BASE_IMAGE_VERSION=latest
+FROM ghcr.io/music-assistant/base:$BASE_IMAGE_VERSION AS builder
 
-FROM ghcr.io/music-assistant/base:$BASE_IMAGE_VERSION
+# Fastest way to install uv
+COPY --from=ghcr.io/astral-sh/uv:latest /uv /uvx /bin/
+
+# create venv which will be copied to the final image
+ENV VIRTUAL_ENV=/app/venv
+RUN uv venv $VIRTUAL_ENV
 
-ARG MASS_VERSION
-ARG TARGETPLATFORM
 ADD dist dist
+COPY requirements_all.txt .
 
 # pre-install ALL requirements
 # comes at a cost of a slightly larger image size but is faster to start
 # because we do not have to install dependencies at runtime
-COPY requirements_all.txt .
 RUN uv pip install \
-    --no-cache \
+    --find-links "https://wheels.home-assistant.io/musllinux/" \
     -r requirements_all.txt
 
 # Install Music Assistant from prebuilt wheel
+ARG MASS_VERSION
 RUN uv pip install \
     --no-cache \
+    --find-links "https://wheels.home-assistant.io/musllinux/" \
     "music-assistant@dist/music_assistant-${MASS_VERSION}-py3-none-any.whl"
 
+# TODO: delete the unneeded architecture of librespot to decrease size!
```

Comment:
what do you mean by this ?

Reply:
In `music_assistant/server/providers/spotify/bin` there are librespot binaries for different architectures, when building eg an amd64 image the arm ones can be deleted. With the multi-stage build, if we delete them from the `builder` image we'll save some space. Not a huge deal, that's why I left as TODO.

Reply:
ah yeah that is what you mean, well those are max a few MB so maybe not that big of a deal